### PR TITLE
Supervised Transfer Learning: Resnet50

### PR DESCRIPTION
**Pull request by SwimmingPig**
Added 4 files:
- resnet_main.ipynb: This is the main execution code for the resnet transfer learning model.
- resnet_lib.py: This is the logic for the resnet_main file.
- resnet_eval.ipynb: This file prints out the wrong predictions the model make.
- resnet_test_create_dataset.ipynb: This is a file to test the create_dataset() in the resnet_lib file.

### CODE REVIEW

File: Supervised_learning/resnet_lib.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        image, label = self.test_data
+        loss, acc = self.model.evaluate(image, label, verbose=0)
+        print('\nTesting loss: {}, acc: {}\n'.format(loss, acc))
+
+
+# Create transfer learning model from pretrained Resnet50 model.
+def create_model(image_size, num_classes, lr):
+    input_tensor = Input(shape=(image_size, image_size, 3))
+    # Create the base pre-trained model
+    base_model = ResNet50(weights='imagenet', include_top=False)
```
Comment (dheath0) [inline Supervised_learning/resnet_lib.py:58]: Does ResNet50 do any image preproccessing? That you'd do when parsing the input, such as normalizing the pixel values as floats, or subtracting the mean. For example, maybe look here:

https://www.tensorflow.org/api_docs/python/tf/keras/applications/resnet/preprocess_input

File: Supervised_learning/resnet_lib.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        self.test_data = test_data
+
+    def on_epoch_end(self, epoch, logs={}):
+        image, label = self.test_data
+        loss, acc = self.model.evaluate(image, label, verbose=0)
+        print('\nTesting loss: {}, acc: {}\n'.format(loss, acc))
+
+
+# Create transfer learning model from pretrained Resnet50 model.
+def create_model(image_size, num_classes, lr):
```
Comment (dheath0) [inline Supervised_learning/resnet_lib.py:51]: Can you add comments to the function header to explain the input/output types? I know some of these are straightforward, but often with Python it can be hard to tell what "type" you are working with---especially when reading someone else's code.

Also, do you have access to go/pystyle ? Lots of great style advice for python readability.

**Review by dheath0 — COMMENTED** [review]
Looks good, just a couple thoughts

File: Supervised_learning/resnet_lib.py (hunk 2 above)
Comment (SwimmingPig) [reply Supervised_learning/resnet_lib.py:51]: Done. I've added comments for the input/output types for all functions.
I do not have access to go/pystyle, but there is a google python style guide here: https://google.github.io/styleguide/pyguide.html

File: Supervised_learning/resnet_lib.py (hunk 1 above)
Comment (SwimmingPig) [reply Supervised_learning/resnet_lib.py:58]: Thanks for pointing it out. Yes, I should add a data preproccessing layer. I thought it was only for imagenet inputs, but after some searching, it does help for transform learning the model.
I've added the preprocessing unit to the model. I get similar results with and without the preprocessing unit. I will train on a few more trails to see if there is a difference.

Comment (dheath0) [reply Supervised_learning/resnet_lib.py:58]: If you're starting from pretrained weights, then I think it is for sure needed. If training from scratch, it may not be as impactful.

**Review by dheath0 — APPROVED** [review]

**Review by dheath0 — APPROVED** [review]

**Review by dheath0 — COMMENTED** [review]
Approving again
